Allow to add missing file hashes + fixup on README

litchipi opened the pull request:
Hello, thanks for the work on this, it looks great !

I had some trouble using it with [OpenDingux](https://github.com/OpenDingux/buildroot) buildroot setup tho, as they do some non-orthodox stuff in there.
Managed to fixed my issues, and thought this might come handy upstream.

Also, there's a problem in the README, if you don't pass `cp -L` to copy the lockfile (it stays a symlink), you'll get a forbidden as nix will try to access to a file outside the flake.

velentr commented on default.nix:
```diff
@@ -8,6 +8,7 @@
   defconfig,
   lockfile,
   nativeBuildInputs ? [],
+  addShaHashes ? {},
```
Rename to `extraSha256Hashes` to clarify that they must be `sha256`

velentr commented on make-package-lock.py:
```diff
@@ -37,6 +38,11 @@ def create_download_info(
+            if source not in checksums_index:
+                name = package["name"]
+                print(f"Missing source {name} in checksums")
+                print(f"URLs:", "\n".join(uris))
```
`f"URLs:"` doesn't need to be an f-string

velentr reviewed (commented):
Looks good with a couple of minor comments.

I generally like to set my commit titles to `<filename>: [attribute]: Title`, would you mind amending your commits to follow this? Would be something like:
```
default.nix: packageLockFile: Allow to add missing file hashes + fixup on README
```
And:
```
make-package-lock.py: Add error in case a checksum isn't found for a specific package
```
You should also add yourself as a contributor to these files with `reuse annotate --contributor`.

litchipi committed:
readme: Add contributor Litchi Pi + add -L flag in buildroot.lock example

Signed-off-by: Litchi Pi <litchi.pi@proton.me>

litchipi committed:
default.nix: packageLockFile: Allow to add missing file hashes

Signed-off-by: Litchi Pi <litchi.pi@proton.me>

litchipi commented:
Addressed the changes, added myself as a contributor, and created 2 clean commits from it (one for readme, one for the extraSha256Hashes)
Good catch, thanks for the review
